File: core/x/javaapi/codegen/src/test/resources/golden/expected/com/example/grts/Query_OrdersConnection_Arguments.java
package com.example.grts;

import graphql.schema.GraphQLInputObjectType;
import viaduct.java.api.context.ExecutionContext;
import viaduct.java.api.globalid.GlobalID;
import viaduct.java.api.reflect.CompositeField;
import viaduct.java.api.reflect.Field;
import viaduct.java.api.reflect.Type;
import viaduct.java.api.reflect.TypeFields;
import java.time.Instant;
import java.time.LocalDate;
import java.time.OffsetTime;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;
import viaduct.apiannotations.InternalApi;
import viaduct.java.api.types.Arguments;
import viaduct.java.api.internal.InputBase;
import viaduct.java.api.internal.InternalContext;

/** Generated arguments class for resolver field. */
@SuppressWarnings("MissingOverride")
public class Query_OrdersConnection_Arguments extends InputBase implements Arguments, viaduct.java.api.types.ForwardConnectionArguments {

    public static final Type<Query_OrdersConnection_Arguments> Reflection = Type.ofClass(Query_OrdersConnection_Arguments.class);

    public static final class Fields implements TypeFields<Query_OrdersConnection_Arguments> {
        private Fields() {}

        public static final Field<Query_OrdersConnection_Arguments> __typename =
                Field.of("__typename", Reflection);
                public static final Field<Query_OrdersConnection_Arguments> first =
                                Field.of("first", Reflection);

    }

    // Public because the framework constructs arguments reflectively across packages
    // (JavaFieldResolverExecutorImpl, VariablesProviderExecutorImpl, etc.). @InternalApi
    // marks it as not-for-tenant-use, mirroring Kotlin's `internal constructor`.
    @InternalApi
    public Query_OrdersConnection_Arguments(InternalContext context, Map<String, Object> data, GraphQLInputObjectType graphQLInputObjectType) {
        super(context, data, graphQLInputObjectType);
    }

    /**
     * Returns whether this input contains a value for {@code field} after GraphQL
     * defaults are applied. Explicit {@code null} counts as present. An omitted field
     * with a schema default is present; an omitted field without a default is absent.
     */
    public boolean isPresent(Field<Query_OrdersConnection_Arguments> field) {
        return isFieldPresent(field);
    }

        public Integer getFirst() {
            return get("first");
        }

        public String getAfter() {
            return null;
        }


    public static Builder builder(ExecutionContext context) {
        return new Builder(InternalContext.from(context));
    }

    public static class Builder {
        private final InternalContext __context;
        private final GraphQLInputObjectType graphQLInputObjectType;
        private final Map<String, Object> data = new LinkedHashMap<>();

        private Builder(InternalContext __context) {
            this.__context = __context;
            this.graphQLInputObjectType =
                    __context.getArgumentsInputType(
                            "Query_OrdersConnection_Arguments",
                            "Query",
                            "ordersConnection");
        }

                public Builder first(Integer first) {
                    data.put("first", first);
        return this;
                }


        public Query_OrdersConnection_Arguments build() {
            return new Query_OrdersConnection_Arguments(
                    __context, new LinkedHashMap<>(data), graphQLInputObjectType);
        }
    }
}

File: core/tenant/codegen/src/test/resources/golden/expected/grts/Query_OrdersConnection_Arguments.kt
@file:Suppress("warnings")

package viaduct.api.grts

import graphql.schema.GraphQLInputObjectType
import viaduct.apiannotations.InternalApi
import viaduct.api.context.ExecutionContext
import viaduct.api.internal.InputTypeFactory
import viaduct.api.internal.InputValueBuilder
import viaduct.api.internal.InternalContext
import viaduct.api.internal.internal
import viaduct.api.internal.InputLikeBase
import viaduct.api.types.Input

@OptIn(InternalApi::class)
class Query_OrdersConnection_Arguments internal constructor(
    override val context: InternalContext,
    override val inputData: Map<String, Any?>,
    override val graphQLInputObjectType: GraphQLInputObjectType,
): InputLikeBase(), viaduct.api.types.Arguments, viaduct.api.types.ForwardConnectionArguments {
    init {
       TODO()
    }

            final override val first: kotlin.Int get() = TODO()

            final override val after: kotlin.String? get() = TODO()


    fun toBuilder() = Builder(context, graphQLInputObjectType, this.inputData.toMutableMap())

    object of {
        operator fun invoke(context: ExecutionContext, block: Builder.() -> Unit): Query_OrdersConnection_Arguments =
            Builder(context).apply(block).build()
    }

    class Builder internal constructor(
        override val context: InternalContext,
        override val graphQLInputObjectType: GraphQLInputObjectType,
        override val inputData: MutableMap<String, Any?> = TODO()
    ) : InputLikeBase.Builder(), InputValueBuilder<Query_OrdersConnection_Arguments> {

        constructor(context: ExecutionContext): this(
            context.internal,
            InputTypeFactory.argumentsInputType("Query_OrdersConnection_Arguments", "Query", "ordersConnection", context.internal.schema),
            mutableMapOf()
        )

        init {
            TODO()
        }

                    fun first(value: kotlin.Int): Builder = TODO()

                    fun after(value: kotlin.String?): Builder = TODO()


        final override fun build(): Query_OrdersConnection_Arguments = TODO()
    }

    @OptIn(viaduct.apiannotations.InternalApi::class)
    object Reflection : viaduct.api.reflect.Type<viaduct.api.grts.Query_OrdersConnection_Arguments> {
        override final val name = "Query_OrdersConnection_Arguments"
        override final val kcls = viaduct.api.grts.Query_OrdersConnection_Arguments::class
    }
    object Fields : viaduct.api.reflect.TypeFields<viaduct.api.grts.Query_OrdersConnection_Arguments> {
            final val __typename: viaduct.api.reflect.Field<viaduct.api.grts.Query_OrdersConnection_Arguments> =
                viaduct.api.internal.FieldImpl("__typename", viaduct.api.grts.Query_OrdersConnection_Arguments.Reflection)

            final val first: viaduct.api.reflect.Field<viaduct.api.grts.Query_OrdersConnection_Arguments> =
                viaduct.api.internal.FieldImpl("first", viaduct.api.grts.Query_OrdersConnection_Arguments.Reflection)

            final val after: viaduct.api.reflect.Field<viaduct.api.grts.Query_OrdersConnection_Arguments> =
                viaduct.api.internal.FieldImpl("after", viaduct.api.grts.Query_OrdersConnection_Arguments.Reflection)

    }

}